<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>WireGuard</class>
 <widget class="QDialog" name="WireGuard">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>758</width>
    <height>766</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>WireGuard Configuration</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QGridLayout" name="gridLayout">
     <item row="1" column="0">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Client PublicKey:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Client IP:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QLineEdit" name="WgClientIPLineEdit">
       <property name="toolTip">
        <string>Desired IP in VPN for this instance. Needs to be agreed upon with the server.</string>
       </property>
       <property name="inputMask">
        <string>000.000.000.000;_</string>
       </property>
       <property name="text">
        <string>192.168.123.</string>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QLineEdit" name="WgClientPubKeyLineEdit">
       <property name="toolTip">
        <string>Public key that needs to be communicated to the server to get VPN access.</string>
       </property>
       <property name="text">
        <string>not yet generated...</string>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="0" column="2">
      <widget class="QSpinBox" name="WgServerPortSpinBox">
       <property name="prefix">
        <string>Port: </string>
       </property>
       <property name="maximum">
        <number>99999</number>
       </property>
       <property name="value">
        <number>9450</number>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="WgServerIPLineEdit">
       <property name="toolTip">
        <string>VPN (WireGuard) server's IP.</string>
       </property>
       <property name="inputMask">
        <string>000.000.000.000;_</string>
       </property>
       <property name="text">
        <string>194.218.146.239</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Server IP:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Server PublicKey:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QLineEdit" name="WgServerPubKeyLineEdit">
         <property name="toolTip">
          <string>VPN (WireGuard) server's public key.</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="WgGenConfigPushButton">
       <property name="text">
        <string>Generate Configuration</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="WgLoadConfigPushButton">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="text">
        <string>Load Existing Configuration</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QLabel" name="label_6">
     <property name="text">
      <string>Configuration for review (manual changes possible):</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPlainTextEdit" name="WgConfigPlainTextEdit">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="placeholderText">
      <string>not yet generated...</string>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="WgApplyConfigPushButton">
       <property name="text">
        <string>Apply Configuration</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="WgCancelPushButton">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
